kolibri/core/device/api.py

```diff
@@ -99,3 +102,29 @@ def get(self, request, format=None):
         info['installer'] = installation_type()
 
         return Response(info)
+
+
+class DeviceLanguageSettingViewset(views.APIView):
+
+    permission_classes = (UserHasAnyDevicePermissions, )
+
+    def get(self, request):
+        settings = DeviceSettings.objects.get()
+        return Response({
+            'language_id': settings.language_id
+        })
+
+    def patch(self, request):
+        settings = DeviceSettings.objects.get()
+
+        try:
+            request_lang = request.data.get('language_id', '')
+        except (Exception):
+            request_lang = ''
```

Comment:
Under what circumstances does this happen?

Recommend only catching the specific exception type you're expecting and letting others raise an exception so that errors aren't lost.

Reply:
If the code falls into this branch, the response will be a 400 a few lines below here.

This can happen if the PATCH payload is ill-formed in at least two-ways that I can think of:

1. There's no `language_id` in the payload (translates to KeyError or something)
1. The payload isn't JSON (translates to JSONError or something).

Reply:
In that case I might do something like

```python
try:
    request_lang = request.data.get('language_id', '')
except AttributeError:
    return HttpResponseBadRequest('Value must be a JSON object: {}'.format(request.data))
```

Then if any other exception occurs, it will throw a 500 which would be more appropriate for an unexpected error.

Reply:
also below you're checking for `None`, so maybe this should actually be `request.data.get('language_id')`? (instead of returning an empty string as the default)

Reply:
I kind of wanted to keep the error handling simple since there's only one consumer of this API, and 500 errors crash the app.

It's also not correct to not leave a specific default for getting 'language_id' since this will happen

```
foo_bar = { "foo": "bar" }
ok = { "language_id": None } # None converted from null

foo_bar.get('language_id') == None # Is acceptable when it shouldn't be
```